src/database/utilities/createRecord.js

```diff
@@ -780,8 +782,55 @@ const createRequisitionItem = (database, requisition, item, dailyUsage, stockOnH
     sortIndex: requisition.items.length + 1,
   });
 
+  if (program && type === 'response') {
+    const { startDate, endDate } = period;
+    const { id: realItemId } = realItem;
+
+    const requisitions = database
+      .objects('Requisition')
+      .filtered(
+        'otherStoreName == $0 && program == $1 && status == "finalised" && type == "response"',
+        otherStoreName,
+        program
+      )
+      .sorted([
+        ['period.startDate', true],
+        ['entryDate', true],
+      ]);
+
+    if (requisitions.length) {
+      const { id: requisitionId } = requisitions[0];
+
+      const lastRequisitionItem = database
+        .objects('RequisitionItem')
+        .filtered('requisition.id == $0 && item.id == $1', requisitionId, realItemId);
+
+      if (lastRequisitionItem[0]) requisitionItem.openingStock = lastRequisitionItem[0].stockOnHand;
+    }
+
+    const customerInvoices = database
+      .objects('TransactionBatch')
+      .filtered(
+        "type == 'stock_out' && transaction.otherParty.id == $0 && itemId == $1",
```

Comment:
Technically this is also counting any unfinalised customer requisition lines. This seems slightly odd but I think is just how mobile works

Reply:
Weird I agree, we are calculating and saving a value on a variable this isn't set in stone. But likewise, someone could make a new CI within the period which would equally throw it out. Maybe. New stock movements updating requisitionItems would also be naff!

@craigdrown might have a thought. Should incoming stock calculation for the customer requisition include unfinalised CI lines to the customer? Could put this in the "Users need to keep on top of finalising their work!" bucket.

I'm also wondering about the assumption here that a given customer only receives stock from one supplier (and that supplier happens to be the store on the tablet, so you have the full transaction info). That'll be true in a lot of circumstances, but maybe not all? The alternative means just not doing this though (for lack of being able to sync all transaction data everywhere!), so 🤷‍♂️  deal with it when it comes up haha.

Reply:
> Should incoming stock calculation for the customer requisition include unfinalised CI lines to the customer?

I think this question should be: 'Given we can only calculate incoming stock for the customer through unfinalised CI/CR (Technically, every CR line has a matching CI line) lines to the customer from the mobile tablet' should we bother?

Note that the incoming stock *is* editable, and this is more of a 'here's a value it could be to save you a second, while potentially just confusing people on where this number even came from!'